src/controllers/product.js

```diff
@@ -235,11 +235,56 @@ const orderRequest = async (req, res) => {
   }
 };
 
+// eslint-disable-next-line consistent-return
+const approveProduct = async (req, res) => {
+  try {
+    const product = await ProductModel.findById(req.params.productId);
+    const requester = await UserModel.findById(req.params.requesterId);
+
+    if (!product) {
+      return res.status(404).json({
+        message: 'Product not found',
+      });
+    }
+    const requesterInOrders = product.orderRequests.includes(
+      req.params.requesterId
+    );
+
+    if (!requester || !requesterInOrders) {
+      return res.status(404).json({
+        message: 'Requester not found',
+      });
+    }
+    if (product.postType === 'Donate') {
+      if (!product.isTransactionCompleted) {
+        requester.received.push(product._id);
+        product.beneficiary = requester._id;
+        product.isTransactionCompleted = true;
+
```

Comment:
i think we also need to delete the product id from other requesters' requested array in user model with a loop, but if it is too much work or too complicated for now, we can skip that it is up to you

Reply:
Yes it can be done , using mongoose methods without a loop

Reply:
Done

Reply:
no no i meant also from the other requesters' array the ones that are not approved, but as i said before, not that important rn so we can move on

Reply:
I wrote it like that the first time around, but then I figured it's not important either way if the transaction is completed you can't edit the requesters .
I thought i would leave it to know that this requester was in the list.